test/e2e/framework/docker.go

```
@@ -76,7 +78,30 @@ func (d *Docker) RunCommand(command ...string) (string, string) {
 	cmd.Stderr = &stderr
 
 	err := cmd.Run()
+
+	return stdout.String(), stderr.String(), err
+}
+
+func (d *Docker) RunCommand(command ...string) (string, string) {
+	stdout, stderr, err := d.runCommand(command...)
 	Expect(err).NotTo(HaveOccurred())
 
-	return stdout.String(), stderr.String()
+	return stdout, stderr
+}
+
+func (d *Docker) RunCommandUntil(command ...string) (string, string) {
+	var cmdErr error
+	var stdout, stderr string
+
+	err := wait.PollImmediate(5*time.Second, time.Duration(TestContext.OperationTimeout)*time.Second,
+		func() (bool, error) {
+			stdout, stderr, cmdErr = d.runCommand(command...)
+			if cmdErr != nil {
+				return false, nil // nolint:nilerr // Returning nil value is intentional
+			}
```

I think we still want to report the command error if the retry timeout is reached. You could use `framework.AwaitUntil` or keep track of the last error and report it below.

---

@tpantelis  Thank you for your comments. I've changed it to report the command errors. This aligns to the behavior of [`IsTransientError`](https://github.com/submariner-io/shipyard/blob/2982bca7e57d74891fe9dcdee876e48fcaa1c3ac/test/e2e/framework/api_errors.go#L27-L34) called from  [`AwaitResultOrError`](https://github.com/submariner-io/shipyard/blob/devel/test/e2e/framework/framework.go#L524). Now it outputs errors like below and keep retrying:

```
STEP: Sending an http request from external app "172.117.0.4" to "10.1.192.6" in the cluster "cluster1"
Mar 31 14:42:03.945: INFO: Transient failure when attempting to run [curl -m 3 10.1.192.6:80/e2e-tests-ext-dataplane-kdg9fcluster1]: exit status 28
```

---

OK. My only nit is that it's not known if it's a transient error so perhaps change the output to "_Error attempting to run..._"

---

I agree. Fixed and squashed the commits.